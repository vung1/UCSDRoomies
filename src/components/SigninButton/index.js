import React from "react";
import { View, Text, Image, TouchableOpacity, StyleSheet } from "react-native";
import useAuth from "../../hooks/useAuth";

function SigninButton(props) {

  const { navigation, screenCurr, state} = props;
  const [message, setState] =  React.useState('')
  const reg = /^\w+([\.-]?\w+)*@ucsd.edu/;
  const { logIn, loading } = useAuth();

  return (
    <View> 
      <Text style={styles.errorMsg}>{message}</Text>
      <TouchableOpacity {...props}
        testID="Signin.Button"
        onPress={() => {
          setState("");
          screenCurr == "LoginScreen" ? 
          navigation.navigate("SigninScreen","SigninScreen") : 
          state.email == "" ?
          setState("* Please enter your email") :
          reg.test(state.email) != true ?
          setState("* Please enter your UCSD email in right format") :
          state.password == "" ?
          setState("* Please enter your password") :
          // TODO: Check validation, if profile is completed go to homepage, 
          // else go to profile page
          logIn(state)
          // console.log(state)
        }}
        activeOpacity={0.8}
        style={styles.buttonContainer}
      >
        <Image
          source={require("../../../assets/images/tritonLogo.png")}
          style={styles.image}
        />
        <Text style={styles.buttonText}> SIGN IN WITH UCSD EMAIL </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  errorMsg: {
    fontSize: 12,
    color: "red",
    marginBottom: "5%",
    marginLeft: "5%",
  },
  buttonContainer: {
    height: 55,
    width: "100%",
    borderRadius: 30,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
    flexDirection: "row",
  },
  buttonText: {
    fontWeight: "bold",
    fontFamily: "Arial",
    fontSize: 16,
    color: "black",
    marginRight: 10,
  },
  image: {
    width: 25,
    height: 25,
    marginLeft: 10,
    marginRight: 10,
  },
});

export default SigninButton;
