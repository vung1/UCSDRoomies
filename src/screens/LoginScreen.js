import React from 'react';
import {View, Text, Image, StyleSheet} from 'react-native';
import {LinearGradient} from 'expo-linear-gradient';

import Logo from '../components/Logo'

const LoginScreen = () => {
  return (
    <LinearGradient colors={["#74AED6", "#247DCF"]} style={styles.background}>
      
      <View style={styles.pageContainer}></View><Logo/>
      
      <View style={styles.buttonContainer}>
        <Text style={styles.buttonText}> LOGIN WITH UCSD EMAIL </Text>
      </View>
    </LinearGradient>
  );
}
const styles = StyleSheet.create({
  background: { 
    flex: 1,
    width: '100%',
    alignItems: "center",
  },
  pageContainer: {
    marginTop: "50%",
  },
  buttonContainer: { 
    padding: 10, 
    alignItems: "center", 
    borderRadius: 5 
  },
  buttonText: {
    fontWeight: "bold",
    fontSize: 15,
    color: "black",
  }
});

export default LoginScreen;