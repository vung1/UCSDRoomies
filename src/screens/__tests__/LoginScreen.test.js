/* eslint-disable import/no-extraneous-dependencies */

import React from "react";

import {
  render,
  fireEvent,
  screen,
  act,
  waitFor,
  mount,
} from "@testing-library/react-native";

// import userEvent from "@testing-library/user-event";

import * as renderer from "react-test-renderer";
import { expect } from "@jest/globals";
import LoginScreen from "../LoginScreen";
import SigninButton from "../../components/SigninButton";

describe("Rendering Tests", () => {
  it("renders default elements", async () => {
    await render(<LoginScreen />);
    expect(screen.getAllByText("SIGN IN WITH UCSD EMAIL").length).toBe(1);
    expect(screen.getAllByText("SIGN UP WITH UCSD EMAIL").length).toBe(1);
  });

  test("snapshot matches", () => {
    const tree = renderer.create(<LoginScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Test navigation
describe("Navigation Tests", () => {
  it("should go to SignIn screen when tapped", async () => {
    const navigation = { navigate: () => {} };
    jest.spyOn(navigation, "navigate");
    // render your component
    await render(<LoginScreen navigation={navigation} />);
    // access your button
    const button = screen.getByTestId("Signin.Button");
    // simulate button click
    fireEvent.press(button);
    // expect result
    expect(navigation.navigate).toHaveBeenCalledWith(
      "SigninScreen",
      "SigninScreen",
    );
  });

  it("should go to SignUp screen when tapped", async () => {
    const navigation = { navigate: () => {} };
    jest.spyOn(navigation, "navigate");
    // render your component
    await render(<LoginScreen navigation={navigation} />);
    // access your button
    const button = screen.getByTestId("Signup.Button");
    // simulate button click
    fireEvent.press(button);
    // expect result
    expect(navigation.navigate).toHaveBeenCalledWith(
      "SignupScreen",
      "SignupScreen",
    );
  });
});
