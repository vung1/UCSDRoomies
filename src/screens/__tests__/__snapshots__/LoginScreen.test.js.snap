// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Tests snapshot matches 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4285837014,
        4280581583,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": "42%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 100,
          "height": 200,
          "width": 200,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 90,
            "marginBottom": "3%",
            "marginTop": "8%",
            "width": 90,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "AmericanTypewriter-Bold",
            "fontSize": 24,
          }
        }
      >
         
        UCSD
         
      </Text>
      <Text
        style={
          Object {
            "fontFamily": "AmericanTypewriter-Bold",
            "fontSize": 24,
          }
        }
      >
         
        Roomies
         
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "10%",
        "width": "80%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 30,
          "flexDirection": "row",
          "height": 55,
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="Signin.Button"
    >
      <Image
        source={1}
        style={
          Object {
            "height": 25,
            "marginLeft": 10,
            "marginRight": 10,
            "width": 25,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Arial",
            "fontSize": 16,
            "fontWeight": "bold",
            "marginRight": 10,
          }
        }
      >
         SIGN IN WITH UCSD EMAIL 
      </Text>
    </View>
    <View
      style={
        Object {
          "marginTop": "7%",
        }
      }
    />
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "red",
            "borderRadius": 30,
            "flexDirection": "row",
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="Signup.Button"
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "marginLeft": 10,
              "marginRight": 10,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "black",
              "fontFamily": "Arial",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginRight": 10,
            }
          }
        >
           SIGN UP WITH UCSD EMAIL 
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "marginTop": "8%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontFamily": "Arial",
          "fontSize": 13,
          "fontWeight": "bold",
          "lineHeight": 16,
          "textAlign": "center",
        }
      }
    >
      By clicking Log In, you agree with our Terms. 
      

      Learn how we process your data in our Privacy 
      

      Policy and Cookies Policy.
    </Text>
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
